great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -248,9 +254,76 @@ def _validate_batch_request(self, batch_request: BatchRequest) -> None:
 )  # read_xml doesn't exist for pandas < 1.3
 
 
+class DataFrameAsset(_PandasDataAsset, Generic[_PandasDataFrameT]):
+    # instance attributes
+    type: Literal["dataframe"] = "dataframe"
+    dataframe: _PandasDataFrameT = pydantic.Field(..., exclude=True, repr=False)
+
+    class Config:
+        extra = pydantic.Extra.forbid
+
+    @pydantic.validator("dataframe")
+    def _validate_dataframe(cls, dataframe: pd.DataFrame) -> pd.DataFrame:
+        if not isinstance(dataframe, pd.DataFrame):
+            raise ValueError("dataframe must be of type pandas.DataFrame")
+        return dataframe
+
+    def _get_reader_method(self) -> str:
+        raise NotImplementedError(
+            """DataFrameAsset is not dynamically generated by a Pandas reader method."""
+        )
+
+    def get_batch_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> list[Batch]:
+        self._validate_batch_request(batch_request)
+        batch_list: List[Batch] = []
+
+        batch_spec = RuntimeDataBatchSpec(batch_data=self.dataframe)
+        execution_engine: PandasExecutionEngine = self.datasource.get_execution_engine()
+        data, markers = execution_engine.get_batch_data_and_markers(
+            batch_spec=batch_spec
+        )
+
+        # batch_definition (along with batch_spec and markers) is only here to satisfy a
+        # legacy constraint when computing usage statistics in a validator. We hope to remove
+        # it in the future.
+        # imports are done inline to prevent a circular dependency with core/batch.py
+        from great_expectations.core import IDDict
+        from great_expectations.core.batch import BatchDefinition
+
+        batch_definition = BatchDefinition(
+            datasource_name=self.datasource.name,
+            data_connector_name="experimental",
```

Comment:
`non-blocking`
Can we use a constant for this?
Also, should we change the name to `fluent`?

☝️ these might be better as followups